libtextsecure/message_receiver.js

```diff
@@ -1122,7 +1130,10 @@ MessageReceiver.prototype.extend({
           // This call already removes the envelope from the cache
           await this.handleContacts(envelope, syncMessage.contacts);
           removedFromCache = true;
-          await this.sendFriendRequestsToSyncContacts(syncMessage.contacts);
+          // We need to wait here because initAPIs hasn't been called yet
+          Whisper.events.on('apisReady', async () => {
```

Comment:
Mh what happens if `apisReady` fires before we start processing the sync message?
Maybe we could make `initialisedAPI` visible and do
```js
if (!windows.initialisedAPI) {
   Whisper.events.on('apisReady',...);
} else {
  // call the thing directly
  await this.sendFriendRequestsToSyncContacts(syncMessage.contacts);
}
```
Also, probably `once` should suffice instead of `on`?

Reply:
True, updated